interface/src/lib.rs

```diff
@@ -97,15 +98,133 @@ impl std::fmt::Display for NetworkInRequest {
     }
 }
 
+#[derive(CandidType, Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Txid([u8; 32]);
+
+impl AsRef<[u8]> for Txid {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl serde::Serialize for Txid {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        serializer.serialize_bytes(&self.0)
+    }
+}
+
+impl<'de> serde::de::Deserialize<'de> for Txid {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        struct TxidVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for TxidVisitor {
+            type Value = Txid;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a 32-byte array")
+            }
+
+            fn visit_bytes<E>(self, value: &[u8]) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                match TryInto::<[u8; 32]>::try_into(value) {
+                    Ok(txid) => Ok(Txid(txid)),
+                    Err(_) => Err(E::invalid_length(value.len(), &self)),
+                }
+            }
+        }
+
+        deserializer.deserialize_bytes(TxidVisitor)
+    }
+}
```

Comment:
Can you not avoid this by having the `#[serde_bytes(with = "serde_bytes")]` annotation in the type definition?

Reply:
I guess you want to be able to return custom errors when deserialization fails but then maybe we use `serde(deserialize_with)` and pass a function that does the checks you want.

Reply:
serde_bytes cannot handle byte arrays.

Reply:
Oh no... Ok, then.

Reply:
I also considered https://docs.rs/serde_arrays/latest/serde_arrays/, but these can't handle blobs.